hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClient.java

```diff
@@ -128,4 +139,30 @@ public void close(long pollInterval, long timeout) throws InterruptedException,
         }
     }
 
+    private void handleMaxRetriesExceeded(ExecutionCompletedEvent<HermesResponse> event) {
+        if (event.getResult().isSuccess()) {
+            return;
+        }
+
+        HermesMessage message = event.getResult().getHermesMessage();
+        if (shouldPublishMetrics()) {
+            String prefix = MetricsUtils.getMetricsPrefix(message.getTopic());
+            metrics.counter(prefix + ".failure.unsent").inc();
+        }
+
+        LOGGER.error(MessageFormat.format("Failed to send message to topic {0} after {1} attempts",
```

Comment:
Do we really need `MessageFormat`? `slf4j` supports parametrised logging. (https://www.slf4j.org/faq.html#logging_performance)